app/src/test/java/org/stepic/droid/util/StringExtensionsTest.kt

```diff
@@ -0,0 +1,43 @@
+package org.stepic.droid.util
+
+import org.junit.Test
+import org.junit.Assert.assertEquals
+
+
+class StringExtensionsTest {
+
+    @Test
+    fun countWhileTest() {
+        val testString = "a \n    x    \n b"
+        assertEquals(
+                testString.indexOf('x') - testString.indexOf('a') - 1, // whitespace symbols between 'a' and 'x'
+                testString.countWhile(testString.indexOf('x') - 1, reversed = true, predicate = Character::isWhitespace))
+
+        assertEquals(
+                testString.indexOf('b') - testString.indexOf('x') - 1, // whitespace symbols between 'x' and 'b'
+                testString.countWhile(testString.indexOf('x') + 1, predicate = Character::isWhitespace))
+    }
+
+    @Test
+    fun substringOrNullTest() {
+        val testString = "1232321"
+
+        assertEquals(null, testString.substringOrNull(-1, 4))
+        assertEquals(null, testString.substringOrNull(5, 4))
+        assertEquals(null, testString.substringOrNull(0, testString.length + 1))
+        assertEquals(testString, testString.substringOrNull(0, testString.length))
```

Comment:
Please, split it into multiple tests. 
When test will failed, a developer should fix 1st assert, after that 2nd assert can be failed, a developer will fix 2nd assert, after that the problem might be in 3rd assert, etc